File: M/896-monotonic-array/main.go
package main

import "fmt"

func isMonotonic(A []int) bool {
	if len(A) <= 1 {
		return true
	}

	direction := 0
	for i := 0; i < len(A)-1; i++ {
		if A[i] == A[i+1] {
			continue
		} else {
			if direction == 0 {
				if A[i] < A[i+1] {
					direction = 1
				} else {
					direction = -1
				}
			} else {
				if A[i] < A[i+1] && direction == -1 {
					return false
				}
				if A[i] > A[i+1] && direction == 1 {
					return false
				}
			}
		}
	}
	return true
}

func main() {
	fmt.Println("a")
}

File: M/896-monotonic-array/main.rs
struct Solution {}

impl Solution {
  pub fn is_monotonic(a: Vec<i32>) -> bool {
    if a.len() <= 2 {
      return true;
    }
    let mut isFirst = true;
    let mut isAsc = true;
    for i in 1..a.len() {
      if isFirst {
        if a[i] > a[i - 1] {
          isAsc = true;
          isFirst = false;
        } else if a[i] < a[i - 1] {
          isAsc = false;
          isFirst = false;
        }
      } else {
        if a[i] == a[i - 1] {
          continue;
        }
        if isAsc != (a[i] > a[i - 1]) {
          return false;
        }
      }
    }
    true
  }
}

fn main() {
  println!("{}", Solution::is_monotonic(vec![0; 5]));
}
